### ISSUE TITLE
[wasm-browser] "Waiting for tests failed" failure

### ISSUE THREAD

**Issue by directhex** [dotnet/runtime#41269]
Some test suites fail with the message "waiting for tests failed"

```
[14:10:16] dbug: test-browser[0]
      Waiting for tests failed
System.AggregateException: One or more errors occurred. (The HTTP request to the remote WebDriver server for URL http://localhost:43993/session/79ab4477fb4b704ac60897ac3e80a42d/element timed out after 60 seconds.)
 ---> OpenQA.Selenium.WebDriverException: The HTTP request to the remote WebDriver server for URL http://localhost:43993/session/79ab4477fb4b704ac60897ac3e80a42d/element timed out after 60 seconds.
 ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
 ---> System.IO.IOException: Unable to read data from the transport connection: Operation canceled.
 ---> System.Net.Sockets.SocketException (125): Operation canceled
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
   --- End of inner exception stack trace ---
   at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebDriver.FindElement(String mechanism, String value)
   at OpenQA.Selenium.By.<>c__DisplayClass22_0.<Id>b__0(ISearchContext context)
   at OpenQA.Selenium.By.FindElement(ISearchContext context)
   at OpenQA.Selenium.Remote.RemoteWebDriver.FindElement(By by)
   at Microsoft.DotNet.XHarness.CLI.Commands.Wasm.WasmBrowserTestRunner.<>c.<RunTestsWithWebDriver>b__7_1(IWebDriver e) in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs:line 74
   at OpenQA.Selenium.Support.UI.DefaultWait`1.Until[TResult](Func`2 condition)
   at Microsoft.DotNet.XHarness.CLI.Commands.Wasm.WasmBrowserTestRunner.<>c__DisplayClass7_0.<RunTestsWithWebDriver>b__0() in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs:line 73
   at System.Threading.Tasks.Task`1.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   --- End of inner exception stack trace ---
[14:10:18] crit: test-browser[0]
      System.AggregateException: One or more errors occurred. (The HTTP request to the remote WebDriver server for URL http://localhost:43993/session/79ab4477fb4b704ac60897ac3e80a42d/element timed out after 60 seconds.)
       ---> OpenQA.Selenium.WebDriverException: The HTTP request to the remote WebDriver server for URL http://localhost:43993/session/79ab4477fb4b704ac60897ac3e80a42d/element timed out after 60 seconds.
       ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
       ---> System.IO.IOException: Unable to read data from the transport connection: Operation canceled.
       ---> System.Net.Sockets.SocketException (125): Operation canceled
         --- End of inner exception stack trace ---
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
         at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
         --- End of inner exception stack trace ---
         at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
         at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
         at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
         at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
         --- End of inner exception stack trace ---
         at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
         at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
         at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
         at OpenQA.Selenium.Remote.RemoteWebDriver.FindElement(String mechanism, String value)
         at OpenQA.Selenium.By.<>c__DisplayClass22_0.<Id>b__0(ISearchContext context)
         at OpenQA.Selenium.By.FindElement(ISearchContext context)
         at OpenQA.Selenium.Remote.RemoteWebDriver.FindElement(By by)
         at Microsoft.DotNet.XHarness.CLI.Commands.Wasm.WasmBrowserTestRunner.<>c.<RunTestsWithWebDriver>b__7_1(IWebDriver e) in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs:line 74
         at OpenQA.Selenium.Support.UI.DefaultWait`1.Until[TResult](Func`2 condition)
         at Microsoft.DotNet.XHarness.CLI.Commands.Wasm.WasmBrowserTestRunner.<>c__DisplayClass7_0.<RunTestsWithWebDriver>b__0() in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs:line 73
         at System.Threading.Tasks.Task`1.InnerInvoke()
         at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
      --- End of stack trace from previous location where exception was thrown ---
         at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
         at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
         --- End of inner exception stack trace ---
         at Microsoft.DotNet.XHarness.CLI.Commands.Wasm.WasmBrowserTestRunner.RunTestsWithWebDriver(IWebDriver driver)
         at Microsoft.DotNet.XHarness.CLI.Commands.Wasm.WasmTestBrowserCommand.InvokeInternal(ILogger logger) in /_/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:line 51
         at Microsoft.DotNet.XHarness.Common.CLI.Commands.XHarnessCommand.Invoke(IEnumerable`1 arguments) in /_/src/Microsoft.DotNet.XHarness.Common/CLI/Commands/XHarnessCommand.cs:line 119
XHarness exit code: 1001
```

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.IO.Compression.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Memory.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Numerics.Vectors.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Runtime.Numerics.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Runtime.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Text.Encodings.Web.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Text.Json.Tests/console

https://helix.dot.net/api/2019-06-17/jobs/300c957c-4b0f-4b89-8ea4-ae7720bfa7c1/workitems/System.Xml.Xsl.XslTransformApi.Tests/console

**Comment by radical** [comment]
I think that the browser is probably crashing here. Selenium seems to timeout while trying to get a response for `FindElement`.

**Comment by radical** [comment]
The browser isn't crashing. For some unknown reason, Chromedriver stops responding to `GetLog` commands (`session/{sessionId}/se/log`), and so the socket ends up timing out. I tried to get to debug that, or get it to work, but no luck.
So, instead I tried using CDP to connect to this session, and listen to the console logging events. That .. works, almost! The test runner just stops receiving messages after some number. The tests are still running because I can see the messages in the *chrome* debug log. So, I'm not sure what is causing this.
Worst case, we can get the output from that debug log, or some other hacky way :/

WIP!
